examples/gno.land/p/demo/forms/validate.gno

```diff
@@ -0,0 +1,150 @@
+package forms
+
+import (
+	"strings"
+
+	"gno.land/p/demo/json"
+)
+
+func validateBooleanField(node *json.Node, field Field) bool {
+	if node.IsBool() == false {
+		return false
+	}
+
+	answer, err := node.GetBool()
+	if err != nil {
+		return false
+	}
+
+	// If the field is required, checkbox must be checked
+	if field.Required == true && answer == false {
+		return false
+	}
+
+	return true
+}
+
+func validateStringField(node *json.Node, field Field) bool {
+	if node.IsString() == false {
+		return false
+	}
+
+	answer, err := node.GetString()
+	if err != nil {
+		return false
+	}
+
+	// If the field is required, the answer must not be empty
+	if field.Required == true && answer == "" {
+		return false
+	}
+
+	return true
+}
+
+func validateNumberField(node *json.Node, field Field) bool {
+	if node.IsNumber() == false {
+		return false
+	}
+
+	_, err := node.GetNumeric()
+	if err != nil {
+		return false
+	}
+
+	return true
+}
+
+func validateMultiChoiceField(node *json.Node, field Field) bool {
+	choices := strings.Split(field.FieldType[1:len(field.FieldType)-1], "|")
+
+	if node.IsArray() == false {
+		return false
+	}
+
+	if field.Required == true && node.Size() == 0 {
+		return false
+	}
+
+	if node.Size() > len(choices) {
+		return false
+	}
+
+	for i := 0; i < node.Size(); i++ {
+		choiceNode, err := node.GetIndex(i)
+		if err != nil {
+			return false
+		}
+
+		choiceIdx := choiceNode.MustNumeric()
+		if choiceIdx < 0 || int(choiceIdx) >= len(choices) {
+			return false
+		}
+	}
+
+	return true
+}
+
+func validateChoiceField(node *json.Node, field Field) bool {
+	choices := strings.Split(field.FieldType[1:len(field.FieldType)-1], "|")
+
+	if node.IsNumber() == false {
+		return false
+	}
+
+	choiceIdx := node.MustNumeric()
+	if choiceIdx < 0 || int(choiceIdx) >= len(choices) {
+		return false
+	}
+
+	return true
+}
+
+// ValidateAnswers checks if the given answers are valid for the given fields
+func ValidateAnswers(answers string, fields []Field) bool {
```

Comment:
It would be good to extract the field validation and choice type checking part into separate functions.